### PR TITLE
[CTC] Fix undefined behavior

### LINKED ISSUE #873: [test/ctc] Code quality: undefined-behavior
ctc.cpp:130: The ctc_alpha_cpu function contains write access to an array by negative offset. This code leads to undefined behavior and should be fixed.

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/test/ctc.cpp#L130-L144

To reproduce:
```
CXX=/opt/rocm/llvm/bin/clang++ CXXFLAGS='-Werror'  cmake -DMIOPEN_TEST_FLAGS=' --disable-verification-cache ' -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS_DEBUG='-g -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=undefined ' -DBUILD_DEV=On .. 
make check
```

```
...
/MLOpen/test/ctc.cpp:130:28: runtime error: addition of unsigned offset to 0x000001e4b630 overflowed to 0x000001e4b628
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /MLOpen/test/ctc.cpp:130:28 in 
CMake Error at test_test_ctc.cmake:7 (message):
...
```

### PR DESCRIPTION
**Pull request by ce1adon**
Resolves #873

### CODE REVIEW

**Review by atamazov — APPROVED** [review]
LGTM

File: test/ctc.cpp [hunk 1]
```diff
@@ -121,12 +121,15 @@ void ctc_alpha_cpu(std::vector<int>& probsDesc,
...
-            int lb_pre      = label_prime[i - 2];
+            int lb_cur = label_prime[i];
+            int lb_pre = -1;
+            if(i >= 2)
+                lb_pre      = label_prime[i - 2];
             size_t pidx     = j * probs_stride[0] + batch_id * probs_stride[1] + lb_cur;
             size_t aidx_ts  = j * label_prime_len + i;
             size_t aidx_t1s = (j - 1) * label_prime_len + i;
 
+            assert(aidx_t1s >= 2);
```
Comment (atamazov) [inline test/ctc.cpp:132]: assertion failed at Fp32 Hip Debug CI stage.

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
@ce1adon Please have a look at CI, thanks.

File: test/ctc.cpp (hunk 1 above)
Comment (ce1adon) [reply test/ctc.cpp]: Thanks! Fixed.

**Review by shurale-nkn — APPROVED** [review]
LGTM

**Review by atamazov — APPROVED** [review]

**Pull request merged**